core/src/test/java/io/onetable/client/TestTableFormatClientFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.onetable.client;
+
+import static io.onetable.GenericTable.getTableName;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mockito;
+
+import io.onetable.iceberg.IcebergColumnStatsConverter;
+import io.onetable.iceberg.IcebergPartitionSpecExtractor;
+import io.onetable.iceberg.IcebergPartitionSpecSync;
+import io.onetable.iceberg.IcebergSchemaExtractor;
+import io.onetable.iceberg.IcebergSchemaSync;
+import io.onetable.model.storage.TableFormat;
+import io.onetable.model.sync.SyncMode;
+import io.onetable.spi.sync.TargetClient;
+
+public class TestTableFormatClientFactory {
+  @TempDir private static java.nio.file.Path tempDir;
+  private final IcebergSchemaExtractor mockSchemaExtractor =
+      Mockito.mock(IcebergSchemaExtractor.class);
+  private final IcebergPartitionSpecExtractor mockPartitionSpecExtractor =
+      Mockito.mock(IcebergPartitionSpecExtractor.class);
+  private final IcebergSchemaSync mockSchemaSync = Mockito.mock(IcebergSchemaSync.class);
+  private final IcebergPartitionSpecSync mockPartitionSpecSync =
+      Mockito.mock(IcebergPartitionSpecSync.class);
+  private final IcebergColumnStatsConverter mockColumnStatsConverter =
+      Mockito.mock(IcebergColumnStatsConverter.class);
+
+  @Test
+  public void testTableClientFromNameForDELTA() {
+    TargetClient tc =
+        TableFormatClientFactory.getInstance().createTargetClientForName(TableFormat.DELTA);
+    assertNotNull(tc);
+    PerTableConfig perTableConfig =
+        getPerTableConfig(Arrays.asList(TableFormat.DELTA), SyncMode.INCREMENTAL);
+    Configuration conf = new Configuration();
+    conf.setStrings("spark.master", "local");
+    tc.init(perTableConfig, conf);
+    assertNotNull(tc);
+  }
```

Comment:
This test only checks whether the target client was created successfully, not whether it is a delta client.